src/components/App/App.js

```diff
@@ -25,140 +20,138 @@ import EditUser from '../EditUser/EditUser';
-              // with authRedirect:
-              // - if logged in, redirects to "/user"
-              // - else shows RegisterPage at "/registration"
-              exact
-              path="/registration"
-              component={RegisterPage}
-              authRedirect="/user"
-            />
-            <ProtectedRoute
-              // with authRedirect:
-              // - if logged in, redirects to "/user"
-              // - else shows LandingPage at "/home"
-              exact
-              path="/home"
-              component={LandingPage}
-              authRedirect="/user"
-            />
+          <ProtectedRoute
+            // logged in shows UserHub else shows LoginPage
+            exact
+            path="/userhub"
+            component={UserHub}
+          />
+          <ProtectedRoute
+            // logged in shows EditUser else shows LoginPage
+            exact
+            path="/edituser"
+            component={EditUser}
+          />
+          <ProtectedRoute
+            // logged in shows EditDeck else shows LoginPage
+            exact
+            path="/editdeck/:id"
+            component={EditDeck}
+          />
 
-            {/* If none of the other routes matched, we will show a 404. */}
-            <Route render={() => <h1>404</h1>} />
-          </Switch>
-          <Footer />
-        </div>
+          <ProtectedRoute
+          // NOT CURRENTLY USING THIS ROUTE
+          exact
+          path="/asyncinput"
+          component={Asyncinput}
+          />
+          
+          {/* When a value is supplied for the authRedirect prop the user will
+          be redirected to the path supplied when logged in, otherwise they will
+          be taken to the component and path supplied. */}
+          <ProtectedRoute
+            // with authRedirect:
+            // - if logged in, redirects to "/user"
+            // - else shows LoginPage at /login
+            exact
+            path="/login"
+            component={LoginPage}
+            authRedirect="/userHub"
+          />
+          <ProtectedRoute
+            // with authRedirect:
+            // - if logged in, redirects to "/userHub"
+            // - else shows RegisterPage at "/registration"
+            exact
+            path="/registration"
+            component={RegisterPage}
+            authRedirect="/userHub"
+          />
+          <ProtectedRoute
+            // with authRedirect:
+            // - if logged in, redirects to "/userHub"
+            // - else shows LandingPage at "/home"
+            exact
+            path="/home"
+            component={LandingPage}
+            authRedirect="/userHub"
+          />
+          <ProtectedRoute
+            exact
+            path="/userDecks"
+            component={UserDecks}
+            authRedirect="/userHub"
+          />
+          {/* If none of the other routes matched, we will show a 404. */}
+          <Route render={() => <h1>404</h1>} />
```

Comment:
I'm not sure 404 is the best error to show. It is correct from an error code meaning "not found", but in the UI it is best to try to give the user a better error message. Like "the page you are trying to navigate to doesn't exist". Then provide links back to other pages. Or just redirect to /login. Then if they are logged in it will automatically redirect then to their user page.